Crashlytics/Crashlytics/Rollouts/CrashlyticsRemoteConfigManager.swift

```diff
@@ -0,0 +1,61 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import FirebaseRemoteConfigInterop
+import Foundation
+
+protocol CrashlyticsPersistentLog: NSObject {
+  func updateRolloutsStateToPersistence(rolloutAssignments: [RolloutAssignment])
+}
+
+@objc(FIRCLSRemoteConfigManager)
+public class CrashlyticsRemoteConfigManager: NSObject {
+  public static let maxRolloutAssignment = 128
+  public static let maxParameterValueLength = 256
+
+  var remoteConfig: RemoteConfigInterop
+  public private(set) var rolloutAssignment: [RolloutAssignment] = []
+  weak var persistenceDelegate: CrashlyticsPersistentLog?
+
+  @objc public init(remoteConfig: RemoteConfigInterop) {
+    self.remoteConfig = remoteConfig
+  }
+
+  @objc public func updateRolloutsState(rolloutsState: RolloutsState) {
+    rolloutAssignment = validateRolloutAssignment(assignments: Array(rolloutsState.assignments))
+  }
+}
+
+private extension CrashlyticsRemoteConfigManager {
+  func validateRolloutAssignment(assignments: [RolloutAssignment]) -> [RolloutAssignment] {
```

Comment:
Slight nit: I usually use validate to refer to checking if it's valid (eg. a function that would return true / false), but not actually modifying it. Could we maybe call it `sanitize` or `normalizeRolloutAssignment` to indicate to the reader it is truncated?

Reply:
Ah I see, I think on Android side I called it `validate` so just port over the same name here. Maybe should fix Android side later 😅

Reply:
Done for the change!